piff/simplepsf.py

```diff
@@ -0,0 +1,223 @@
+
+        :param config_psf:      The psf field of the configuration dict, config['psf']
+        :param logger:          A logger object for logging debug info. [default: None]
+
+        :returns: a kwargs dict to pass to the initializer
+        """
+        import piff
+
+        kwargs = {}
+        kwargs.update(config_psf)
+
+        for key in ['model', 'interp']:
+            if key not in kwargs:
+                raise ValueError("%s field is required in psf field for type=Simple"%key)
+
+        # make a Model object to use for the individual stellar fitting
+        model = piff.Model.process(kwargs.pop('model'), logger=logger)
+        kwargs['model'] = model
+
+        # make an Interp object to use for the interpolation
+        interp = piff.Interp.process(kwargs.pop('interp'), logger=logger)
+        kwargs['interp'] = interp
+
+        if 'outliers' in kwargs:
+            outliers = piff.Outliers.process(kwargs.pop('outliers'), logger=logger)
+            kwargs['outliers'] = outliers
+
+        return kwargs
+
+    def fit(self, stars, wcs, pointing,
+            chisq_threshold=0.1, max_iterations=30, logger=None):
+        """Fit interpolated PSF model to star data using standard sequence of operations.
+
+        :param stars:           A list of Star instances.
+        :param wcs:             A dict of WCS solutions indexed by chipnum.
+        :param pointing:        A galsim.CelestialCoord object giving the telescope pointing.
+                                [Note: pointing should be None if the WCS is not a CelestialWCS]
+        :param chisq_threshold: Change in reduced chisq at which iteration will terminate.
+                                [default: 0.1]
+        :param max_iterations:  Maximum number of iterations to try. [default: 30]
+        :param logger:          A logger object for logging debug info. [default: None]
+        """
+        # TODO: Make chisq_thresh and max_iterations configurable paramters and move them
+        #       to the initialization.
+        self.stars = stars
+        self.wcs = wcs
+        self.pointing = pointing
+
+        if logger:
+            logger.debug("Initializing models")
+        self.stars = [self.model.initialize(s, mask=True, logger=logger) for s in self.stars]
+
+        if logger:
+            logger.debug("Initializing interpolator")
+        self.stars = self.interp.initialize(self.stars, logger=logger)
+
+        # For basis models, we can compute a quadratic form for chisq, and if we are using
+        # a basis interpolator, then we can use it.  It's kind of ugly to query this, but
+        # the double dispatch makes it tricky to implement this with class heirarchy, so for
+        # now we just check if we have all the required parts to use the quadratic form
+        quadratic_chisq = hasattr(self.model, 'chisq') and self.interp.degenerate_points
+
+        # Begin iterations.  Very simple convergence criterion right now.
```

Comment:
> If the PSF model doesn't have refluxing and outlier rejection, then the below for loop will run either 1 time (fit is below threshold) or max_iterations times (fit is above threshold but there is no way to change the fit since there is no refluxing or outlier rejection).

Not quite.  The test is on `oldchisq - chisq`.  So if there are no changes, this will be 0, which will test as less than the (positive) `chisq_threshold * dof`.  So it will stop on the second iteration.

I think models _should_ always have a `reflux` function.  Certainly, all relatively sophisticated models will want to have this to handle the case of missing data (masked pixels).  The best estimate for those missing pixels comes from the interpolated model.  

The current check about `if hasattr(self.model, 'reflux')` is kind of a hack to avoid the issue for now, since I didn't implement something appropriate for `Gaussian`. And in practice we'll almost always want to use some kind of outlier rejection.  So I think I'm ok with not having an explicit check for the case where there is no mechanism for chisq to decrease.  It's a pretty minor inefficiency for an unusual use case.